From 283e4f63668852bbb690fbfd04b8b9fdbfcd3c6d Mon Sep 17 00:00:00 2001
From: Wu Zhangben <wuzhangben@eswincomputing.com>
Date: Tue, 26 Aug 2025 12:48:47 +0800
Subject: [PATCH 490/541] fix:power off USB in s3

Changelogs:
1. Power off USB(dwc3-eswin) in s3 state
2. Add 'power-gpios' entry to dts

Signed-off-by: Wu Zhangben <wuzhangben@eswincomputing.com>
---
 arch/riscv/boot/dts/eswin/eic7702-d560-interleave.dts         | 4 +---
 .../boot/dts/eswin/eic7702-deepcomputing-fml13v03-evt.dts     | 1 +
 arch/riscv/boot/dts/eswin/eic7702-som560-interleave.dtsi      | 1 +
 3 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/arch/riscv/boot/dts/eswin/eic7702-d560-interleave.dts b/arch/riscv/boot/dts/eswin/eic7702-d560-interleave.dts
index e6bf44c31648..9e161d76d36a 100644
--- a/arch/riscv/boot/dts/eswin/eic7702-d560-interleave.dts
+++ b/arch/riscv/boot/dts/eswin/eic7702-d560-interleave.dts
@@ -102,9 +102,7 @@ mdhd_wlan {
 };
 
 &d0_usbdrd3_1 {
-	status = "okay";
-	pinctrl-names = "default";
-	pinctrl-0 = <&pinctrl_die0_gpio14_default>;
+	pinctrl-0 = <&pinctrl_die0_usb1_pwren_default>, <&pinctrl_die0_gpio14_default>;
 	hub-rst-gpios = <&porta 14 GPIO_ACTIVE_HIGH>;
 };
 
diff --git a/arch/riscv/boot/dts/eswin/eic7702-deepcomputing-fml13v03-evt.dts b/arch/riscv/boot/dts/eswin/eic7702-deepcomputing-fml13v03-evt.dts
index 6a52df025d21..e7558437cc9e 100755
--- a/arch/riscv/boot/dts/eswin/eic7702-deepcomputing-fml13v03-evt.dts
+++ b/arch/riscv/boot/dts/eswin/eic7702-deepcomputing-fml13v03-evt.dts
@@ -975,6 +975,7 @@ &d1_sata {
 &d0_usbdrd3_0 {
 	status = "okay";
 };
+
 &d1_usbdrd3_0 {
 	status = "okay";
 };
diff --git a/arch/riscv/boot/dts/eswin/eic7702-som560-interleave.dtsi b/arch/riscv/boot/dts/eswin/eic7702-som560-interleave.dtsi
index e19d0ad973a6..26df2a59cca8 100644
--- a/arch/riscv/boot/dts/eswin/eic7702-som560-interleave.dtsi
+++ b/arch/riscv/boot/dts/eswin/eic7702-som560-interleave.dtsi
@@ -879,6 +879,7 @@ &d0_usbdrd3_0 {
 	status = "okay";
 	rt_equalization;
 };
+
 &d1_usbdrd3_0 {
 	status = "okay";
 	rt_equalization;
-- 
2.54.0

